torch/_inductor/ir.py

```diff
@@ -5250,6 +5250,75 @@ def should_allocate(self):
         return False
 
 
+class TMADescriptor(ExternKernel):
+    """
+    An IR node representing a host-side TMA descriptor in the Triton API
+    (the ones obtained via create_{1d,2d}_tma_descriptor calls). Mostly
+    useful for user-defined Triton kernels relying on host-side TMA; but
+    can, in principle, be used for Inductor's Triton templates, too.
+    """
+
+    # as TMA descriptors are immutable,
+    # we can dedup them by the input args
+    _CACHE: Dict[Any, TMADescriptor] = {}
+
+    @classmethod
+    def create(
+        cls,
+        tensor: TensorBox,
+        dims: List[Union[int, torch.SymInt]],
+        block_dims: List[Union[int, torch.SymInt]],
+        element_size: Optional[int] = None,
+    ):
+        key = (id(tensor), dims, block_dims, element_size)
```

Comment:
@eellison I'm using `id(tensor)` here because `TensorBox` happens to be non-hashable. Although this looks correct, but it's very restrictive: we require the same `TensorBox` Python object to hit the cache. Is there a more canonical way to do this in Inductor IR (that would allow, e.g., different `TensorBox`es referring to the same underlying storage)? Maybe I should unwrap storage before doing `id(...)`, or can this ignore offsets in the views which can lead to different `data_ptr()` values?

Reply:
You could try to cover Tensors with same strides, same underlying storage, but even then, would require us to fix the layout. I think this is good.